File: client/classes/prop.js
class Prop {
    /**
     * @param {{ x: number, y: number }} cords
     * @param {{ width: number, height: number }} size
     **/
    constructor(cords, size) {
        this.cords = cords;
        this.size = size;
    }

    /**
     * @param {CanvasRenderingContext2D} ctx 
     **/
    draw(ctx) {
        ctx.fillRect(this.cords.x, this.cords.y, this.size.width, this.size.height);
    }
}

export default Prop;

File: client/src/prop.cpp
#include "prop.h"

Prop::Prop(Coords cords, Size size) : cords(cords), size(size) {
        shape.setSize(sf::Vector2f(size.width, size.height));
        shape.setFillColor(sf::Color::Blue);
        shape.setPosition(cords.x, cords.y);
    }

void Prop::draw(sf::RenderWindow& window) {
    window.draw(shape); 
}
